IDEMPIERE-4863-1 Enable save column width in WListBox

gsjeffen opened the pull request:
This is a new PR after the first review which was messed up during the update after the review by me.

This PR is the base for implementing save of column width of a wlistbox in forms and windows.

hengsin commented on org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/WListbox.java:
```diff
@@ -1255,5 +1277,100 @@ public void setAllowIDColumnForReadWrite(boolean allowIDColumnForReadWrite)
+				custom.deleteCharAt(custom.length() - 1);
+			ok = MWlistboxCustomization.saveData(Env.getCtx(), getwListBoxName(), Env.getAD_User_ID(Env.getCtx()), custom.toString(), null);
+		}			
+		return ok;
```
Should be better to change the method's return type to void and have the saveData method throws exception if there's error.

gsjeffen replied:
Done

hengsin commented on org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/WListbox.java:
```diff
@@ -1255,5 +1277,100 @@ public void setAllowIDColumnForReadWrite(boolean allowIDColumnForReadWrite)
+     * Render the header of the WListbox with the columns width of the saved column width
+     * 
+     */
+	public void renderHeaderColumnWidth () 
```
this perhaps doesn't needs to be public and it is more obvious to name this as renderCustomHeaderWidth() instead.

gsjeffen replied:
keeping it as public, but changing the name to renderCustomHeaderWidth as it makes more sense.

hengsin commented on org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/WListbox.java:
```diff
@@ -1255,5 +1277,100 @@ public void setAllowIDColumnForReadWrite(boolean allowIDColumnForReadWrite)
+     */
+	public boolean saveColumnWidth() 
+	{
+		boolean ok = false;
```
this method should check getwListBoxName() is not empty

gsjeffen replied:
check

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,138 @@
+			WlistBoxCust.setCustom(Custom);
+		}
+		if (Util.isEmpty(WlistBoxCust.getCustom(), true)) {
+			return WlistBoxCust.delete(true);
```
It is a weird flow that we did all the process above for update/create and then only do the delete here.

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,138 @@
+			return WlistBoxCust.delete(true);
+		}
+		return WlistBoxCust.save();
+	}  // saveData
```
better to change method return type to void and call saveEx() and deleteEx() instead.

gsjeffen replied:
Changed

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,138 @@
+				
+				for (int i = 0; i < newColumnList.size(); i++)
+				{
+					String searchColumnName = newColumnList.get(i).toString().substring(0,newColumnList.get(i).toString().indexOf("="));
```
you don't have to call toString() for String type.

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,138 @@
+					else
+					{ 
+						addColumn.add(newColumnList.get(i).toString().stripLeading());
+					}
```
again, it is strange that we keep calling toString() for String type object.

gsjeffen replied:
Removed

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,138 @@
+		Query query = new Query(ctx, Table_Name, "AD_User_ID=? AND WListboxName=?", trxName);
+		return query.setClient_ID().setOnlyActiveRecords(true).setParameters(new Object[] { AD_User_ID, AD_WListboxName}).first();
+	}
+
```
please add basic javadoc for public constructor and methods and remove all the //TODO tag.

hengsin commented on migration/i8.2z/postgresql/202106281200_IDEMPIERE-4863.sql:
```diff
@@ -0,0 +1,115 @@
+;
+
+	
+SELECT register_migration_script('202106281200_IDEMPIERE-4863.sql') FROM dual;	
```
this is not a must but it would be better if you can use current date for migration script.

gsjeffen replied:
Ok, noted.

hengsin commented on org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/WListbox.java:
```diff
@@ -1255,5 +1277,100 @@ public void setAllowIDColumnForReadWrite(boolean allowIDColumnForReadWrite)
+					{
+						Listheader header = (Listheader) obj;
+						String columnWidth = this.getColumnWidth(Columns[i].getColumnName(), columnList);
+						if (!("null".equals(columnWidth)) || !Util.isEmpty(columnWidth))	
```
shouldn't this be "&&" instead ? also, the "&&" or "||" is not needed if the getColumnWidth(...) method just return null instead.

gsjeffen replied:
You are right.

hengsin commented on org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/WListbox.java:
```diff
@@ -1255,5 +1277,100 @@ public void setAllowIDColumnForReadWrite(boolean allowIDColumnForReadWrite)
+				return w[1];
+			}
+		}
+		return width;
```
should be simpler to just return null here and Util.isEmpty(...) below will works for both null and empty string.

gsjeffen replied:
Util.isEmpty(...) return 0 or 1. The null value is save in the custom field in the database and is used to show that we haven't been able to get or save any column width.

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,138 @@
+		return new ArrayList<String>(Arrays.asList(CustomizationNew)); 
+	}	
+
+	private static int columnExists(String 				searchColumnName, 
```
more accurate to name this as "columnIndex" instead.

hengsin commented on migration/i8.2z/oracle/202106281200_IDEMPIERE-4863.sql:
```diff
@@ -0,0 +1,96 @@
+    UpdatedBy 						NUMBER(10) NOT NULL,
+    AD_Wlistbox_Customization_UU 	VARCHAR2(36 CHAR) DEFAULT NULL,
+	CONSTRAINT AD_Wlistbox_Customization_Key PRIMARY KEY (AD_Wlistbox_Customization_ID),
+	CONSTRAINT WlistboxName_UU UNIQUE (AD_Client_ID,AD_User_ID,WlistboxName),
```
sound weird to name the constraint as _UU as that can confuse people that the constraint is for the "AD_Wlistbox_Customization_UU" column instead.

gsjeffen reviewed (commented):
I have updated the PR with the purposed changes.

/Flemming

hengsin commented on migration/i8.2z/oracle/202106281200_IDEMPIERE-4863.sql:
```diff
@@ -0,0 +1,96 @@
+);
+
+
+CREATE UNIQUE INDEX ad_wlistbox_customization_uu_idx ON ad_wlistbox_customization (ad_wlistbox_customization_uu);
```
The naming issue is still not address: ad_wlistbox_customization_uu_idx is for ad_wlistbox_customization_uu and AD_Wlistbox_Customization_UU constraint for (AD_Client_ID,AD_User_ID,WlistboxName). Should use a different name for the  AD_Wlistbox_Customization_UU unique constraint.

gsjeffen replied:
Ok, done.

gsjeffen reviewed (commented):
Updated constraint

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,152 @@
+		{
+			CustomizationNew[i] = CustomizationNew[i].substring(0, CustomizationNew[i].indexOf("=") );
+		}
+		return new ArrayList<String>(Arrays.asList(CustomizationNew)); 
```
The more concise ways is to use return type of List<String> and you can remove the use of new ArrayList<String>(...) here (and should be just List<String> columnSearch below.)

gsjeffen replied:
done

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,152 @@
+		{
+			String[] custOrg = WlistBoxCust.getCustom().split(",");
+			ArrayList<String> newColumnList = new ArrayList<>( Arrays.asList(Custom.trim().split(",")) );
+			ArrayList<String> orgColumnList  = new ArrayList<>( Arrays.asList(WlistBoxCust.getCustom().trim().split(",")) );
```
the use of `new ArrayList<>(...)` here is unnecessary, use List<String> instead.

gsjeffen replied:
Done

hengsin replied:
hi @gsjeffen , why was this being changed back to the "new ArrayList" approach ?

gsjeffen replied:
Hi @hensin

It is only orgColumnList that is changed

List<String> newColumnList =  Arrays.asList(Custom.trim().split(","));
ArrayList<String> orgColumnList  = new ArrayList<>( Arrays.asList(WlistBoxCust.getCustom().trim().split(",")) );

This is due to the following:
orgColumnList.addAll(addColumn);

if orgColumnList is of list then this would give an exception.

hengsin commented on org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java:
```diff
@@ -0,0 +1,152 @@
+			if (!orgColumnList.equals(newColumnList))
+			{
+				ArrayList<String> columnSearch = cleanCustomization (custOrg);
+				ArrayList<String> addColumn = new ArrayList<String>();
```
both declaration here can be changed to `List<String>`

gsjeffen replied:
Done

gsjeffen reviewed (commented):
Updated with review

hengsin commented:
Hi @CarlosRuiz-globalqss ,

This looks like a nice addition to me, wdyt ?

Regards,
Low

CarlosRuiz-globalqss commented:
Hi @gsjeffen, is always better to generate the scripts from iDempiere directly, the scripts here have some problems that are controlled by iDempiere (for example the CHAR notation on oracle), also the index must be created in dictionary.

Please apply the patch [pr883_1.txt](https://github.com/idempiere/idempiere/files/7235384/pr883_1.txt) which solves those problems and also update the X_ class with latest changes.

I think the ticket comes from a forum thread, or from another ticket, but there is no cross-reference about that.

@hengsin - yes, this looks interesting but I don't know how to test it, if you have tested it please go ahead and push it.

Regards,

Carlos Ruiz

gsjeffen commented:
@CarlosRuiz-globalqss & @hengsin 
To be honest, I give up and do throws the towel in the ring! This is probably the best for the project and I will not be wasting your valuable time. 😊
/Flemming

CarlosRuiz-globalqss commented:
> @CarlosRuiz-globalqss & @hengsin
> To be honest, I give up and do throws the towel in the ring! This is probably the best for the project and I will not be wasting your valuable time. blush
> /Flemming

@gsjeffen - sorry to hear that, we want to make the path to contribute easier, but at the same time the most important is to keep the quality of the project.  I understand this process is complicated at the beginning, but after you get it, it becomes easy.

If you have suggestions that can help to ease contributions without sacrificing quality, please let us know.

It was not my intention to annoy you, but the migration scripts break several rules required in oracle and postgresql.

I know the effort you have put on this, so instead of asking you to fix that, I sent you a patch that fixes the script problems found.

If you can't integrate the fix, I guess I can integrate it, but honestly I don't know how to test it, or what is the purpose of this, as @hengsin knows better, that's why I asked him to go ahead if he consider this worthy (but still the script patch is required as we need the database to be good quality).

Regards,

Carlos Ruiz

gsjeffen commented:
@CarlosRuiz-globalqss, actually I have learned a lot today and thanks for that, it is really appreciated. I do understand the thorough review process is needed in order to keep the quality top notch, history has shown you that it was/is necessary. I have applied the patch you created and yes that made it easy to fix.

hengsin commented:
Hi @CarlosRuiz-globalqss ,

This is just the base/framework to save the current column width of a WListbox instance so you can't test it as it is. I guess we can build an implementation for info window to test this out (using "AD_InfoWindow|<AD_InfoWindow_UU>" as wlistboxname" and perhaps add a user preference or sysconfig to say whether to auto save info window column width for user). @gsjeffen , if you want, feel free to go ahead to add the info window implementation (please do restrain from adding anything that's not related to this ticket though).

Regards,
Low

CarlosRuiz-globalqss commented:
@hengsin - if you are OK with that, I think is enough to push it as it is.

gsjeffen commented:
Hi @hengsin

 

Shall I create a new PR for this ticket or should I create a new ticket and then a new PR for the implementation part of infoWindows?

 

/Flemming

 

Fra: hengsin ***@***.***> 
Sendt: 29. september 2021 04:01
Til: idempiere/idempiere ***@***.***>
Cc: gsjeffen ***@***.***>; Mention ***@***.***>
Emne: Re: [idempiere/idempiere] IDEMPIERE-4863-1 Enable save column width in WListBox (#883)

 

Hi @CarlosRuiz-globalqss <https://github.com/CarlosRuiz-globalqss>  ,

This is just the base/framework to save the current column width of a WListbox instance so you can't test it as it is. I guess we can build an implementation for info window to test this out (using "AD_InfoWindow|<AD_InfoWindow_UU>" as wlistboxname" and perhaps add a user preference or sysconfig to say whether to auto save info window column width for user). @gsjeffen <https://github.com/gsjeffen>  , if you want, feel free to go ahead to add the info window implementation (please do restrain from adding anything that's not related to this ticket though).

Regards,
Low

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/idempiere/idempiere/pull/883#issuecomment-929759242> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOCBVVML3MJUV3XPV4V74XLUEJXM3ANCNFSM5EFJX25A> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .   <https://github.com/notifications/beacon/AOCBVVPCONBE7QPYHYVTTW3UEJXM3A5CNFSM5EFJX25KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOG5VQACQ.gif>

hengsin commented:
Hi @gsjeffen ,

You can add it to this PR. Just be extra careful to keep the changes to the minimum possible.

Regards,
Low

hengsin commented:
hi @gsjeffen , why do you remove the org.adempiere.base.Model annotation at c579c897e6b8b5a8db350939d57a19b48a22d022 ?

gsjeffen commented:
Hi @hensin

 

It is only orgColumnList that is changed

 

List<String> newColumnList =  Arrays.asList(Custom.trim().split(","));

ArrayList<String> orgColumnList  = new ArrayList<>( Arrays.asList(WlistBoxCust.getCustom().trim().split(",")) );

 

This is due to the following:

orgColumnList.addAll(addColumn);

 

if orgColumnList is of list then this would give an exception.

 

/Flemming

 

Fra: hengsin ***@***.***> 
Sendt: 4. oktober 2021 07:32
Til: idempiere/idempiere ***@***.***>
Cc: gsjeffen ***@***.***>; Mention ***@***.***>
Emne: Re: [idempiere/idempiere] IDEMPIERE-4863-1 Enable save column width in WListBox (#883)

 

@hengsin commented on this pull request.

  _____  

In org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java <https://github.com/idempiere/idempiere/pull/883#discussion_r721053452> :

> +      * @return
+        */
+       
+       public static void saveData(Properties  ctx,
+                                                            String      AD_WListboxName, 
+                                                            int         AD_User_ID, 
+                                                            String      Custom, 
+                                                            String      trxName) 
+       {       
+              MWlistboxCustomization WlistBoxCust = get(ctx, AD_User_ID, AD_WListboxName, trxName);
+       
+              if (WlistBoxCust != null && WlistBoxCust.getAD_Wlistbox_Customization_ID()> 0)
+              {
+                      String[] custOrg = WlistBoxCust.getCustom().split(",");
+                      ArrayList<String> newColumnList = new ArrayList<>( Arrays.asList(Custom.trim().split(",")) );
+                      ArrayList<String> orgColumnList  = new ArrayList<>( Arrays.asList(WlistBoxCust.getCustom().trim().split(",")) );

hi @gsjeffen <https://github.com/gsjeffen>  , why was this being changed back to the "new ArrayList" approach ?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/idempiere/idempiere/pull/883#discussion_r721053452> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOCBVVKYI7ZJJRPHVXA3ETTUFE36XANCNFSM5EFJX25A> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .   <https://github.com/notifications/beacon/AOCBVVNWQ6S3FKQKU3HYJ6DUFE36XA5CNFSM5EFJX25KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOFXRW6HI.gif>

hengsin commented:
Hi @gsjeffen ,

Well, I guess you have tested it although that shouldn't be the case here ...

Regards,
Low

gsjeffen commented:
Hi @hengsin

 

Yes, i have tested this and I will later today include example implementation of this in infoPanel and infoProductWindow as agreed last week with Carlos. Hope it is ok to append it to this PR?

 

/Flemming

 

Fra: hengsin ***@***.***> 
Sendt: 4. oktober 2021 12:06
Til: idempiere/idempiere ***@***.***>
Cc: gsjeffen ***@***.***>; Mention ***@***.***>
Emne: Re: [idempiere/idempiere] IDEMPIERE-4863-1 Enable save column width in WListBox (#883)

 

Hi @gsjeffen <https://github.com/gsjeffen>  ,

Well, I guess you have tested it although that shouldn't be the case here ...

Regards,
Low

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/idempiere/idempiere/pull/883#issuecomment-933333897> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOCBVVKERVMJFV7VN7B74X3UFF37DANCNFSM5EFJX25A> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .   <https://github.com/notifications/beacon/AOCBVVMAJCMMLAML4PSPTDDUFF37DA5CNFSM5EFJX25KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOG6QYXCI.gif>

hengsin commented:
Hi @gsjeffen ,

Yes, please add it to this PR.

Regards,
Low

hengsin commented on org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java:
```diff
@@ -618,6 +619,7 @@ protected void prepareTable (ColumnInfo[] layout,
         String sql =contentPanel.prepareTable(layout, from,
                 where,p_multipleSelection,
                 getTableName(),false);
+        contentPanel.setwListBoxName("AD_InfoWindow_UU|"+ infoWindow.getAD_InfoWindow_UU() );
```
this is not safe, infoWindow can be null here.

hengsin commented on org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java:
```diff
@@ -2637,7 +2640,7 @@ public void dispose(boolean ok)
     	if (log.isLoggable(Level.CONFIG)) log.config("OK=" + ok);
         m_ok = ok;
-
+        saveWlistBoxColumnWidth(this.getFirstChild());
```
here we should check that wlistboxname is not empty/null

gsjeffen replied:
This is done in the function saveColumnWidth in the class WListbox, so nothing is save without a valid wlistboxname.

hengsin commented on org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/WListbox.java:
```diff
@@ -1255,5 +1277,99 @@ public void setAllowIDColumnForReadWrite(boolean allowIDColumnForReadWrite)
+			{
+				Listheader header = (Listheader) obj;
+				width = header.getWidth();
+				colName = layout[i].getColumnName();
```
needs to check whether columnName is null/empty here.

gsjeffen replied:
Done

gsjeffen commented:
@hengsin this is done in the function saveColumnWidth in the class WListbox, so nothing is save without a valid wlistboxname.

 

/Flemming

 

Fra: hengsin ***@***.***> 
Sendt: 5. oktober 2021 07:51
Til: idempiere/idempiere ***@***.***>
Cc: gsjeffen ***@***.***>; Mention ***@***.***>
Emne: Re: [idempiere/idempiere] IDEMPIERE-4863-1 Enable save column width in WListBox (#883)

 

@hengsin commented on this pull request.

  _____  

In org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java <https://github.com/idempiere/idempiere/pull/883#discussion_r721907956> :

> @@ -2637,7 +2640,7 @@ public void dispose(boolean ok)
     {
        if (log.isLoggable(Level.CONFIG)) log.config("OK=" + ok);
         m_ok = ok;
-
+        saveWlistBoxColumnWidth(this.getFirstChild());

here we should check that wlistboxname is not empty/null

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/idempiere/idempiere/pull/883#pullrequestreview-771001814> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOCBVVKNSFYPT3I2YUHJN6DUFKG3JANCNFSM5EFJX25A> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .   <https://github.com/notifications/beacon/AOCBVVPBMY6XFNFVZXDOATDUFKG3JA5CNFSM5EFJX25KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOFX2I3VQ.gif>

gsjeffen commented:
@hengsin ok I wasn’t aware of this, so should a check be added for checking if infoWindow is not null ?

 

/Flemming

 

Fra: hengsin ***@***.***> 
Sendt: 5. oktober 2021 07:50
Til: idempiere/idempiere ***@***.***>
Cc: gsjeffen ***@***.***>; Mention ***@***.***>
Emne: Re: [idempiere/idempiere] IDEMPIERE-4863-1 Enable save column width in WListBox (#883)

 

@hengsin commented on this pull request.

  _____  

In org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java <https://github.com/idempiere/idempiere/pull/883#discussion_r721907620> :

> @@ -618,6 +619,7 @@ protected void prepareTable (ColumnInfo[] layout,
         String sql =contentPanel.prepareTable(layout, from,
                 where,p_multipleSelection,
                 getTableName(),false);
+        contentPanel.setwListBoxName("AD_InfoWindow_UU|"+ infoWindow.getAD_InfoWindow_UU() );

this is not safe, infoWindow can be null here.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/idempiere/idempiere/pull/883#pullrequestreview-771001395> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOCBVVOPH36PSIAY6QDGVZ3UFKGX5ANCNFSM5EFJX25A> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .   <https://github.com/notifications/beacon/AOCBVVKFECGFPSGQH4QXKUTUFKGX5A5CNFSM5EFJX25KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOFX2IYMY.gif>

hengsin commented:
hi @gsjeffen , yes, please add the null check for info window. should add the null/empty check for columnName too.
P/S, please reply at the "Files changed" tab instead. Your reply here making it difficult to follow the context of your comments.

hengsin commented:
Hi @gsjeffen ,
Please add https://pastebin.pl/view/7a45ba1f to the oracle script and https://pastebin.pl/view/19f23f8c to the postgresql script.

The committed sql script is missing the unique index for ad_wlistbox_customization(ad_client_id,ad_user_id,wlistboxname)

Regards,
Low

gsjeffen commented:
> Hi @gsjeffen ,
> 
> Please apply the patch at https://pastebin.pl/view/643e6f09 and rename the migration script to current date (i.e 20211008 or newer) and we are good to go. Using current date for migration script help to ensure we don't runs into sql script dependency issue in future (when we upgrade the iDempiere seed from 8.2 to 9, the migration script is run in the sorted order of the sql script file name).
> 
> The patch above resolve the following issues:
> 
> * Width of column not save if I close the Product Info tab using the close tab button instead of the cancel button at the bottom.
> * After saving of column width, reopen Product Info tab, run query. It works fine but now if I adjust the width of some of the columns and run query again, it goes back to the previous save width.
> 
> A short coming here is there's no way for user to reset width back to the defaults.
> 
> Regards, Low

gsjeffen commented:
@hengsin I have updated with the patch supplied and renamed the sql file names.
